### PR TITLE
Refactor `astropy/coordinates/tests/test_solar_system.py`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

Writing many similar unit tests leads to code duplication, and writing a unit test with many assertions prevents the later assertions from being executed if an early assertion fails during testing. In both cases writing parametrized unit tests is a good solution. I've tried to parametrize tests in `astropy/coordinates/tests/test_solar_system.py` and managed to reduce the number of lines of code quite a bit.

Also worth highlighting is the new `test_ephemeris_non_existing_url()`, which checks what the code does if the user specifies an URL for ephemerides that can't be found. This has been tested by using an invalid URL, but there had been problems with the query timing out instead of producing the expected HTTP 404 error. In https://github.com/astropy/astropy/commit/e3808d43d8c09c3ad7a80501a9cf7eeb034ac6dd the test was changed by involving `astropy.utils.data.get_pkg_data_filename()` and allowing for an `URLError`, but it was not obvious if the test was still checking what it was supposed to (see the discussion in #12936). The new test guarantees that a 404 error is raised through monkeypatching.

### CODE REVIEW

**Comment by pllim** [comment]
`test_leap_second.py` is related?

**Comment by eerovaher** [comment]
The failures were somehow caused by the second to last commit, so I've removed the last two commits from the pull request for now. I'll try to find out what was it exactly that was causing the errors.

**Comment by eerovaher** [comment]
Some of the tests were parametrized using `Time` instances that apparently invoked the UTC conversion mechanism, which the following code does not like: https://github.com/astropy/astropy/blob/140051b0691c8fd1cc4bc4860c59803a96485d95/astropy/utils/iers/tests/test_leap_second.py#L21-L26
I decided to simply undo the parametrization of those tests. I think this pull request is good enough as it is.

**Comment by pllim** [comment]
I don't grok the original purpose of the tests, so I'd rather coordinates maintainers review this one.

**Comment by StuartLittlefair** [comment]
I can approve the coordinate tests @pllim - the original intent of the tests are unchanged and the code is much neater. 

However, I don't understand the monkey patching code, so if you are content with that, we can merge.

**Review by StuartLittlefair — APPROVED** [review]

File: astropy/coordinates/tests/test_solar_system.py [hunk 1]
```diff
@@ -503,58 +349,75 @@ def test_url_or_file_ephemeris(time):
...
 @pytest.mark.skipif(not HAS_JPLEPHEM, reason="requires jplephem")
-def test_url_ephemeris_wrong_input():
-    time = Time("1960-01-12 00:00")
-    with pytest.raises((HTTPError, URLError)):
-        # A non-existent URL
+def test_ephemeris_non_existing_url(monkeypatch):
+    def request_invalid_url(*args, **kwargs):
+        raise HTTPError(code=404, msg="Not Found", fp=None, hdrs=None, url="")
+
+    monkeypatch.setattr("urllib.request.OpenerDirector.open", request_invalid_url)
```
Comment (pllim) [inline astropy/coordinates/tests/test_solar_system.py:362]: @StuartLittlefair , this monkeypatch makes call to `urllib.request.OpenerDirector.open` raise an error. I think this method is called internally by `get_body` (via `astropy.utils.data`).

Thanks for everyone for the work and the approval. Merging!

**Pull request merged**